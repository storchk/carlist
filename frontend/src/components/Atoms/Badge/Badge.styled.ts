import styled, { css } from 'styled-components'

export const StyledBadge = styled.span(
  ({ theme }) => css`
    display: inline-flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacing.xs} ${theme.spacing.sm};
    background-color: ${theme.colors.contrast200};
    color: ${theme.colors.contrast700};
    border-radius: ${theme.spacing.sm};
  `
)
