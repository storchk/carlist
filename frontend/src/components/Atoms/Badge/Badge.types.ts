export type BadgeColor = 'primary' | 'neutral'
export type BadgeProps = {
  label: string
  color?: BadgeColor
}
