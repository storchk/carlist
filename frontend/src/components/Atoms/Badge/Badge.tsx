import { Typography } from '../Typography'
import { StyledBadge } from './Badge.styled'
import type { BadgeProps } from './Badge.types'
export const Badge = ({ label, ...props }: BadgeProps): JSX.Element => {
  return (
    <StyledBadge {...props}>
      <Typography fontSize={{ mobile: 'xxs' }} fontWeight="bold">
        {label}
      </Typography>
    </StyledBadge>
  )
}
