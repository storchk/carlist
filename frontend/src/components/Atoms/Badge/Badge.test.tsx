import { axe } from 'jest-axe'

import { testing } from '@/testing'
import { Badge } from './Badge'
import type { BadgeProps } from './Badge.types'

const { render, screen, renderer } = testing
const label = 'Lorem Ipsum'

const renderComponent = (props?: Partial<BadgeProps>) => render(<Badge label={label} {...props} />)

describe('atoms: Badge', () => {
  it('should render badge with text', () => {
    renderComponent()
    const text = screen.getByText(/Lorem Ipsum/i)
    expect(text).toBeInTheDocument()
  })

  it('snapshot', () => {
    const view = renderer(<Badge label={label} />)
    expect(view).toMatchSnapshot()
  })
  it('a11y - No violations detected', async () => {
    const { container } = renderComponent()
    await expect(axe(container)).resolves.toHaveNoViolations()
  })
})
