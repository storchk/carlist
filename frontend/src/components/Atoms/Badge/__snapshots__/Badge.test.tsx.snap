// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms: Badge snapshot | color: neutral 1`] = `
.c1 {
  font-weight: 700;
  color: inherit;
  font-size: 12px;
}

.c0 {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 8px 16px;
  border-radius: 16px;
  background-color: #f1f3f9;
  color: #2f3237;
}

<span
  className="c0"
>
  <span
    className="c1"
    fontSize={
      {
        "mobile": "xxs",
      }
    }
    fontWeight="bold"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`atoms: Badge snapshot | color: primary 1`] = `
.c1 {
  font-weight: 700;
  color: inherit;
  font-size: 12px;
}

.c0 {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 8px 16px;
  border-radius: 16px;
  background-color: #f1f3f9;
  color: #2f3237;
}

<span
  className="c0"
>
  <span
    className="c1"
    fontSize={
      {
        "mobile": "xxs",
      }
    }
    fontWeight="bold"
  >
    Lorem Ipsum
  </span>
</span>
`;
